### ISSUE TITLE
feat: display the public IP address of Transmission in UI

### ISSUE THREAD

**Issue by tearfur** [transmission/transmission#5785]
### Discussed in https://github.com/transmission/transmission/discussions/5618

<div type='discussions-op-text'>

<sup>Originally posted by **tearfur** June 14, 2023</sup>
Now that we have the global IP cache implemented (#5329, #5498, #5510), is it a good idea to show in the UI (e.g. in a pop-up opened by a menu item) the public IP that Transmission is being seen as?

I imagine this would be useful for users to troubleshoot connectivity problems, or to ensure that they are binding to the correct interface etc.

Just throwing this out there for discussion.</div>

**Comment by killemov** [comment]
Duplicate of  #1634.

**Comment by ckerr** [comment]
Even though 1634 came first, I'd say it's a subset of this ticket since 1634 is referring only to the web UI. Probably we should close it instead in favor of this one.

**Comment by Pentaphon** [comment]
Recently PR'd at https://github.com/qbittorrent/qBittorrent/pull/20118 in case we need an example

**Comment by Staubgeborener** [comment]
I took a look at the previous PRs and issues here. They keep referring back to this thread and are being closed.

So what is the real reason why this simple feature, which many users have requested, is not being implemented, even though other clients (like the above-mentioned qBittorrent) list it as a default feature?

**Comment by Vaadasch** [comment]
> I took a look at the previous PRs and issues here. They keep referring back to this thread and are being closed.
> 
> So what is the real reason why this simple feature, which many users have requested, is not being implemented, even though other clients (like the above-mentioned qBittorrent) list it as a default feature?

It is opensource, you wanna do it ? There is a "PR welcome" tag

**Comment by Staubgeborener** [comment]
Like I said, there were some PRs which were closed. I don't see the point to create another one which will be closed after some days because it's not wanted. So I politely decline.

**Comment by Vaadasch** [comment]
I saw only this one : https://github.com/transmission/transmission/pull/1664, closed by @ckerr 
Don't really understand why it was closed, i agree.

You said that you find other PR ?

> -1 for relying on an external 3rd party service. There are several torrent leak checkers on the web you can use to validate your configuration.

Comment from @malte-v raised a serious question, as to where retrieve the external IP.
It would be 1000% better to check directly from inside transmission (ipify.org as suggested, or ipecho.net/plain) than having the need to look on google and having one's CPU used for cryptomining by torguard.net...
Either way, the external IP is leaked... because it is the one used to upload and download. Better to not have the real IP leaked wanting to know the external one.

> I'm not sure that I understand the purpose of this feature. What is the use case for being able to see the IP address of the web client?

Reminder, as it seems to be easily misunderstood : The purpose was to know with a single sight the currently used IP address for the torrent connection. Not the IP used by the web client

Transmission is often used behind a vpn, and we need to ensure that it is correctly configured, and that it is not using default interface without vpn.
I think that absolutely everyone check its external IP address after the installation, and regularly during usage. IMHO far more than the current download / upload speed that is displayed

**Comment by maloleroy** [comment]
I agree about the usefulness of this feature, but I'm not sure an external service `ipify.org` would be reliable. A TCP/HTTPS packet might not be routed by the user's ISP and/or VPN the same way as a TCP/BitTorrent packet would, and since a lot of trackers are in UDP/BitTorrent-tracker (with some being in TCP/HTTP), it's even worse. My point is, it's far from obvious to know one's current public IP.

And that is if "public IP" even means anything, because it could be possible for our ISP/VPN to route packets to different endpoints. In a situation where our UDP and TCP packets have different endpoints, we could get pairs through a UDP tracker (with our "UDP public IP") and others through a HTTP tracker (with an entirely different public IP). Assuming we even know these addresses, which one should we display? Probably both, but the point is, networking is tricky.

---

I don't think any external machine except **trackers** and/or connected pairs can reliably know the public IP(s) used. That being said, there _is_ a specification on the BitTorrent protocol regarding trackers providing the external IP: [**BEP 24**](https://www.bittorrent.org/beps/bep_0024.html). As of now it is still a draft, and not standard. It might be implemented by some trackers though, but we can't be certain of that.

---
Edit: my bad I didn't see @tearfur 's PRs. I made a fuss about how to know one's external IP but it's already an implemented feature.

@tearfur you might want to add #8650 to the PRs listed in the opening comment on this issue
